backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/FlatrateDAO.java

```diff
@@ -1213,4 +1214,18 @@ private IQuery<I_ModCntr_Settings> buildModularContractSettingsQueryFilter(@NonN
 
 		return queryBuilder.create();
 	}
+
+	@Override
+	public boolean isExistsModularOrInterimContract(@NonNull final IQueryFilter<I_C_Flatrate_Term> flatrateTermFilter)
+	{
+		return getFlatrateTermQueryBuilder(flatrateTermFilter)
+				.addInArrayFilter(I_C_Flatrate_Term.COLUMNNAME_Type_Conditions, X_C_Flatrate_Term.TYPE_CONDITIONS_ModularContract, X_C_Flatrate_Term.TYPE_CONDITIONS_InterimInvoice)
+				.anyMatch();
+	}
+	@NonNull
+	private IQueryBuilder<I_C_Flatrate_Term> getFlatrateTermQueryBuilder(@NonNull final IQueryFilter<I_C_Flatrate_Term> flatrateTermFilter)
```

Comment:
what is the benefit of this method?

Reply:
I thought this part could be reused later in similar functions, so i extracted it. Like here `de.metas.deliveryplanning.DeliveryPlanningRepository#getDeliveryPlanningQueryBuilder`